lua/lspconfig/configs/gh_actions_ls.lua

```diff
@@ -3,9 +3,16 @@ local util = require 'lspconfig.util'
 return {
   default_config = {
     cmd = { 'gh-actions-language-server', '--stdio' },
-    filetypes = { 'yaml.github' },
-    root_dir = util.root_pattern('.github'),
-    single_file_support = true,
+    filetypes = { 'yaml' },
+
+    -- Only attach to yaml files that are GitHub workflows instead of all yaml
+    -- files. (A nil root_dir and no single_file_support results in the LSP not
+    -- attaching.) For details, see #3558
+    root_dir = function(filename)
+      return filename:find('/%.github/workflows/.+%.ya?ml') and util.root_pattern('.github') or nil
+    end,
+    single_file_support = false,
```

Comment:
```suggestion
    -- Disabling "single file support" is a hack to avoid enabling this LS for
    -- every random yaml file. The `root_dir` definition specifies the kind of 
    -- yaml file that is required for this LS.
    single_file_support = false,
```